trin-core/src/portalnet/overlay.rs

```diff
@@ -175,12 +190,182 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
     }
 
     /// Process overlay uTP listener event
-    pub fn process_utp_event(&self, event: UtpListenerEvent) {
-        // TODO: Handle overlay uTP events
-        debug!(
-            "Got overlay uTP event: {event:?}, protocol id: {:?}",
-            self.protocol
-        );
+    pub fn process_utp_event(&self, event: UtpListenerEvent) -> anyhow::Result<()> {
+        match event {
+            UtpListenerEvent::ClosedStream(utp_payload, _, stream_id) => {
+                // Handle all closed uTP streams, currently we process only AcceptStream here.
+                // FindContent payload is processed explicitly when we send FindContent request.
+                if let AcceptStream(content_keys) = stream_id {
+                    self.process_accept_utp_payload(content_keys, utp_payload)?;
+                }
```

Comment:
These seems to be an easy place to accidentally forget to process something. Maybe a good option would be a match that explicitly does nothing on the `FindContent`. Later, if there's a new entry in the enum, then we get a compiler error here for a missing branch in the match. That way we are forced to consider whether to process it here or not.